src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java

```diff
@@ -128,34 +128,45 @@ public void test() {
-        assertEquals("MERGE_STARTED", messageMergeStarted.getHeaders().get("type"));
-        Message<byte[]> messageMergeFinished = output.receive(1000);
-        assertEquals("MERGE_FINISHED", messageMergeFinished.getHeaders().get("type"));
+        assertEquals("MERGE_PROCESS_STARTED", messageMergeStarted.getHeaders().get("type"));
+
+        Message<byte[]> readNetworksStarted = output.receive(1000);
+        assertEquals("READ_NETWORKS_STARTED", readNetworksStarted.getHeaders().get("type"));
+        Message<byte[]> readNetworksFinished = output.receive(1000);
+        assertEquals("READ_NETWORKS_FINISHED", readNetworksFinished.getHeaders().get("type"));
+
+        Message<byte[]> mergeNetworksStarted = output.receive(1000);
+        assertEquals("MERGE_NETWORKS_STARTED", mergeNetworksStarted.getHeaders().get("type"));
+        Message<byte[]> mergeNetworksFinished = output.receive(1000);
+        assertEquals("MERGE_NETWORKS_FINISHED", mergeNetworksFinished.getHeaders().get("type"));
+        Message<byte[]> mergedNetworksStored = output.receive(1000);
+        assertEquals("MERGED_NETWORK_STORED", mergedNetworksStored.getHeaders().get("type"));
+
         List<MergeEntity> savedFinish = mergeRepository.findAll();
         assertEquals(1, savedFinish.size());
-        assertEquals("MERGE_FINISHED", savedFinish.get(0).getStatus());
+        assertEquals("MERGE_PROCESS_FINISHED", savedFinish.get(0).getStatus());
         assertEquals(mergedUuid, savedFinish.get(0).getNetworkUuid());
         assertEquals("SWE", savedFinish.get(0).getKey().getProcess());
         assertEquals(dateTime, savedFinish.get(0).getKey().getDate());
 
         List<MergeInfos> mergeInfos = mergeOrchestratorService.getMergesList();
         assertEquals(1, mergeInfos.size());
         assertEquals("SWE", mergeInfos.get(0).getProcess());
-        assertEquals("MERGE_FINISHED", mergeInfos.get(0).getStatus());
+        assertEquals("MERGE_PROCESS_FINISHED", mergeInfos.get(0).getStatus());
         assertEquals(dateTime, mergeInfos.get(0).getDate());
 
         mergeInfos = mergeOrchestratorService.getProcessMergesList("SWE");
         assertEquals(1, mergeInfos.size());
         assertEquals("SWE", mergeInfos.get(0).getProcess());
-        assertEquals("MERGE_FINISHED", mergeInfos.get(0).getStatus());
+        assertEquals("MERGE_PROCESS_FINISHED", mergeInfos.get(0).getStatus());
         assertEquals(dateTime, mergeInfos.get(0).getDate());
 
         Optional<MergeInfos> mergeInfo = mergeOrchestratorService.getMerge("SWE", "2019-05-01T10:00:00.000");
         assertTrue(mergeInfo.isPresent());
         assertEquals("SWE", mergeInfo.get().getProcess());
-        assertEquals("MERGE_FINISHED", mergeInfo.get().getStatus());
+        assertEquals("MERGE_PROCESS_FINISHED", mergeInfo.get().getStatus());
         assertEquals(dateTime, mergeInfo.get().getDate());
 
-        assertNull(output.receive(1000));
+        //assertNull(output.receive(1000));
     }
```

Comment:
You can decomment assertNull(output.receive(1000)) here, if you add the test for the messages "LOAD_FLOW_STARTED", "LOAD_FLOW_FINISHED" and "MERGE_PROCESS_FINISHED" before

Reply:
Done